src/helpers.js

```diff
@@ -162,35 +181,20 @@ function registerWith(Handlebars) {
         }
 
         if (typeof message === 'string') {
-            message = getMessageFormat(message, locales, formats);
+            message = getMessageFormat(isHtml ? message : escape(message), locales, formats);
+        }
+
+        if (hasSafeString) {
```

Comment:
If any of the has values is a SafeString, we need to return a SafeString for it to be rendered as HTML.  Everything else is escaped, so shouldn't be a problem.